src/components/views/dialogs/keybackup/CreateKeyBackupDialog.js

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import sdk from '../../../../index';
+import MatrixClientPeg from '../../../../MatrixClientPeg';
+import Promise from 'bluebird';
+
+import { _t, _td } from '../../../../languageHandler';
+
+const PHASE_INTRO = 0;
+const PHASE_GENERATING = 1;
+const PHASE_SHOWKEY = 2;
+const PHASE_MAKEBACKUP = 3;
+const PHASE_UPLOAD = 4;
+const PHASE_DONE = 5;
+
+// XXX: copied from ShareDialog: factor out into utils
+function selectText(target) {
+    const range = document.createRange();
+    range.selectNodeContents(target);
+
+    const selection = window.getSelection();
+    selection.removeAllRanges();
+    selection.addRange(range);
+}
+
+/**
+ * Walks the user through the process of creating an e22 key backup
+ * on the server.
+ */
+export default React.createClass({
```

Comment:
i don't suggest we change this one, but surely we should be creating new stuff as ES6 at last given the language caught up (and React 16 mandates it?)

Reply:
Yeah, I guess we should update the code style as I don't think we have